File: Android/app/src/main/java/ch/pfvr/internapp/RiverDisplay.java
package ch.pfvr.internapp;

/** Station-specific presentation rules for hydrological values. */
final class RiverDisplay {
    private static final double BASEL_RHEINHALLE_GAUGE_ZERO_M = 240.0d;
    private static final double RHEINFELDEN_GAUGE_ZERO_M = 260.0d;

    private RiverDisplay() {}

    static double levelValue(HydroStation station,double rawMetresAboveSea){
        return rawMetresAboveSea;
    }

    static String levelUnit(HydroStation station){
        return "m ü.M.";
    }

    static int levelDecimals(HydroStation station){
        return 2;
    }

    static double gaugeCentimetres(HydroStation station,double rawMetresAboveSea){
        if(!Double.isFinite(rawMetresAboveSea))return rawMetresAboveSea;
        double zero=station==HydroStation.BASEL_RHEINHALLE
                ?BASEL_RHEINHALLE_GAUGE_ZERO_M
                :RHEINFELDEN_GAUGE_ZERO_M;
        return (rawMetresAboveSea-zero)*100.0d;
    }
}

File: iOS/Sources/PFVRCore/RiverDisplay.swift
import Foundation

public enum RiverDisplay {
    public static func hasVerifiedGaugeCentimetres(_ station: HydroStation) -> Bool { station == .baselRheinhalle }
    public static func gaugeCentimetres(station: HydroStation, metresAboveSea: Double) -> Double? {
        guard station == .baselRheinhalle, metresAboveSea.isFinite else { return nil }
        return (metresAboveSea - 240) * 100
    }
    public static func graphLevelValue(station: HydroStation, metresAboveSea: Double, centimetres: Bool) -> Double {
        centimetres ? gaugeCentimetres(station: station, metresAboveSea: metresAboveSea) ?? metresAboveSea : metresAboveSea
    }
    public static func graphLevelUnit(station: HydroStation, centimetres: Bool) -> String {
        centimetres && hasVerifiedGaugeCentimetres(station) ? "cm" : "m ü.M."
    }
    public static func graphLevelDecimals(station: HydroStation, centimetres: Bool) -> Int {
        centimetres && hasVerifiedGaugeCentimetres(station) ? 0 : 2
    }
}
